packages/compat/src/compat-app-builder.ts

```diff
@@ -143,7 +143,11 @@ export class CompatAppBuilder {
   }
 
   private addResolverConfig(config: CompatResolverOptions) {
-    outputJSONSync(join(locateEmbroiderWorkingDir(this.compatApp.root), 'resolver.json'), config, { spaces: 2 });
+    outputJSONSync(
+      join(locateEmbroiderWorkingDir(this.compatApp.root), 'resolver.json'),
+      { ...config, splitAtRoutes: config.splitAtRoutes?.map(s => s.toString()) },
```

Comment:
This is obviously a quick and dirty way to force every regex to a string, but this is essentially relying on regex detection by checking for it starting and ending with `/` when it's being deserialised. I don't know if there is a better way forward for this since we're (currently) relying on this being serialised into JSON 🙈 

It also means that there is currently no way to add this config for people not using a prebuild. Is this the time that we implement some sort of config file outside of the ember-cli-build? Thoughts?

Reply:
I have enhanced the check to see if something is a regex with the help of @mkszepp 's comments, but we still need to decide if this is the right kind of fix 👍